app/controllers/picture_controller.rb

```diff
@@ -0,0 +1,33 @@
+class PictureController < ApplicationController
+  def show # GET /pictures/:basename
+    compressed_id, extension = params[:basename].split('.')
+    picture = Picture.find_by_id(from_cid(compressed_id))
+    if picture && picture.extension == extension
```

Comment:
I just realized...if you are checking the extension directly, then do it in sql?

```ruby
picture = Picture.where(:id => from_cid(compressed_id), :extension => extension)
if picture
  render_picture_content(picture)
else
  render :nothing => true, :status => 404
end
```